src/audio/google/google_rtc_audio_processing.c

```diff
@@ -822,23 +821,60 @@ static int google_rtc_audio_processing_process(struct processing_module *mod,
 	/* 32float: de-interlace ref buffer, convert it to float, skip channels if > Max
 	 * 16int: linearize buffer, skip channels if > Max
 	 */
+	/* Reduce cycle waste by streamlining the inner loop,
+	 * converting from array indexing to pointer arithmetic,
+	 * and putting data copy verification outside the loop.
+	 */
 	buffer_offset = 0;
-	for (int i = 0; i < cd->num_frames; i++) {
-		for (channel = 0; channel < cd->num_aec_reference_channels; ++channel) {
-#if CONFIG_COMP_GOOGLE_RTC_USE_32_BIT_FLOAT_API
-			cd->aec_reference_buffer_ptrs[channel][i] =
-					convert_int16_to_float(ref[channel]);
-#else /* CONFIG_COMP_GOOGLE_RTC_USE_32_BIT_FLOAT_API */
-			cd->aec_reference_buffer[buffer_offset++] = ref[channel];
-#endif /* CONFIG_COMP_GOOGLE_RTC_USE_32_BIT_FLOAT_API */
+	const int16_t *ref_end = ref + cd->num_frames * cd->num_aec_reference_channels;
+
+	if ((void *)ref_end >= (void *)ref_buf_end)
+		ref_end = (void *)ref_buf_start;
```

Comment:
`ref_buf_end`? Also, please don't type-cast to non-`const` types `const` pointers. I'd do
```
	if ((const int8_t *)ref_end > ref_buf_end)
		ref_end = (const int16_t *)ref_buf_end;
```